### PR TITLE
Backup downloadurl

### PR DESCRIPTION
**Pull request by ahmadhamzh**
**What this PR does / why we need it**:
add a button to download backups from the cluster backups page 

[Screencast from 02-14-2024 01:52:33 PM.webm](https://github.com/kubermatic/dashboard/assets/85109141/2aceefe0-5210-4428-a881-0e1eeef00ad3)

**What type of PR is this?**
/kind feature

```release-note
Support for downloading backups from the UI
```

```documentation
TBD
```

### CODE REVIEW

File: modules/web/src/app/core/services/cluster-backup.ts [hunk 1]
```diff
@@ -111,4 +111,9 @@ export class ClusterBackupService {
     const url = `${this._newRestRoot}/projects/${projectID}/clusterbackupstoragelocation/${bslName}`;
     return this._http.patch<BackupStorageLocation>(url, backupStorageLocation);
   }
+
+  postBackupDownloadUrl(projectID: string, clusterID: string, backupName: string): Observable<{downloadURL: string}> {
+    const url = `${this._newRestRoot}/projects/${projectID}/clusters/${clusterID}/clusterbackup/${backupName}/downloadurl`;
+    return this._http.post<{downloadURL: string}>(url, null);
```
Comment (Waseem826) [inline modules/web/src/app/core/services/cluster-backup.ts:117]: Can we create a new Interface for `{downloadURL: string}`?

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve for sig-api approval
/hold
Remove the hold @Waseem826 when everything is good to go

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve
/unhold

**Comment by ahmedwaleedmalik** [comment]
@ahmadhamzh please create a testing ticket for this feature.

**Comment by ahmedwaleedmalik** [comment]
@Waseem826 for code freeze approval

**Comment by Waseem826** [comment]
I've asked @ahmadhamzh for one more change about using `BackupDownloadUrl` interface in `modules/web/src/app/dynamic/enterprise/cluster-backups/list/backups/component.ts` so just waiting for that.

**Comment by ahmedwaleedmalik** [comment]
> I've asked @ahmadhamzh for one more change about using `BackupDownloadUrl` interface in `modules/web/src/app/dynamic/enterprise/cluster-backups/list/backups/component.ts` so just waiting for that.

Apologies for the hasty approval in that case 😁

**Review by Waseem826 — APPROVED** [review]

**Pull request merged**